test/topicFilters.spec.ts

```diff
@@ -0,0 +1,296 @@
+  subscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.SUBSCRIPTION_NAME
+  );
+  defaultSubscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.DEFAULT_SUBSCRIPTION_NAME
+  );
+  await removeAllRules(subscriptionClient);
+}
+
+async function afterEachTest(): Promise<void> {
+  await namespace.close();
+}
+
+describe("Topic Filters - Tests", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  it("Add Rule with Boolean filter", async function(): Promise<void> {
+    await subscriptionClient.addRule("TrueFilter", true);
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "TrueFilter");
+  });
+
+  it("Add Rule with SQL filter", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 2) AND (sys.label LIKE '%String2')"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with SQL filter and action", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+      "SET sys.body = 'MessageX'"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with SQL filter and action", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+      "SET sys.body = 'MessageX'"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with Correlation filter", async function(): Promise<void> {
+    await subscriptionClient.addRule("Correlationfilter", {
+      label: "red",
+      correlationId: "high"
+    });
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Correlationfilter");
+  });
+
+  it("Add Default rule on a subscription that its default rule removed", async function(): Promise<
+    void
+  > {
+    await subscriptionClient.addRule("$Default", {
+      label: "red",
+      correlationId: "high"
+    });
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "$Default");
+  });
+
+  it("Adding a rule with a name which matches with existing rule", async function(): Promise<void> {
+    await subscriptionClient.addRule("Priority_1", "priority = 1");
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "priority = 2");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search(" already exists."), false);
+      console.log(error.message);
```

Comment:
- Doesnt the error message contain the rule name? If so, then can we include that in the search here?
- Does the error have a name? If yes, then lets check for that as well.
- Why do we want to console.log the error?

Same applies to all other test cases where we are expecting an error

Reply:
For the test `"Adding a rule with a Boolean filter whose input is not Boolean"`

> { InternalServerError: The service was unable to process the request; please retry the operation. For more information on exception types and proper exception handling, please refer to http://go.microsoft.com/fwlink/?LinkId=761101, SystemTracker:NoSystemTracker, Timestamp:2019-01-15T19:03:45

Checked many times, this is the error I see. 
Are we expecting something like "Cannot accept this type of expression" ?

Reply:
It has the rule name. Will add that check too.
This is the name of the error: `MessagingEntityAlreadyExistsError`
No need of console.log(), will remove it.

Will do the same at all the places !